pkg/controller/tidbcluster/tidb_cluster_control.go

```diff
@@ -174,6 +185,23 @@ func (c *defaultTidbClusterControl) updateTidbCluster(tc *v1alpha1.TidbCluster)
 		return err
 	}
 
+	// Upgrade TiCDC first if needed.
+	upgradeTiCDCFirst, err := NeedToUpdateTiCDCFirst(ns, tcName, c.deps)
+	if err != nil {
+		return err
```

Comment:
Is it possible to choose one of the upgrade orders if any error exists (some users may use a custom non-semver image tag)?